drivers/clock_control/clock_control_mcux_syscon.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2020, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_lpc_syscon
+#include <errno.h>
+#include <soc.h>
+#include <drivers/clock_control.h>
+#include <dt-bindings/clock/mcux_lpc_syscon_clock.h>
+#include <fsl_clock.h>
+
+#define LOG_LEVEL CONFIG_CLOCK_CONTROL_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(clock_control);
+
+static int mcux_lpc_syscon_clock_control_on(const struct device *dev,
+			      clock_control_subsys_t sub_system)
+{
+	return 0;
+}
+
+static int mcux_lpc_syscon_clock_control_off(const struct device *dev,
+			       clock_control_subsys_t sub_system)
+{
+	return 0;
+}
+
+static int mcux_lpc_syscon_clock_control_get_subsys_rate(
+					const struct device *dev,
+				    clock_control_subsys_t sub_system,
+				    uint32_t *rate)
+{
+#if defined(CONFIG_I2C_MCUX_FLEXCOMM) || \
+		defined(CONFIG_SPI_MCUX_FLEXCOMM) || \
+		defined(CONFIG_UART_MCUX_FLEXCOMM)
+
+	uint32_t clock_name = (uint32_t) sub_system;
+
+	switch (clock_name) {
+	case MCUX_FLEXCOMM0_CLK:
```

Comment:
The case could use  `kCLOCK_FlexComm0..7`, `kCLOCK_Hs_Lspi`, etc.

Reply:
No, the value is coming from device tree so we need to use the macros from `include/dt-bindings/clock/mcux_lpc_syscon_clock.h`